doc/guides/build-system/kconfig.md

```
@@ -0,0 +1,568 @@
+the main output from the Kconfig configuration system. It holds all the macros
+that should be used to configure modules in RIOT:
+`CONFIG_<module>_<parameter>`.
+
+In order to generate the `autoconf.h` file the `genconfig` script is used.
+Inputs for this script are the main `Kconfig` file and `out.config`
+configuration file, which holds the selected values for the exposed parameters.
+
+#### Input:
+- `$ (GENERATED_DIR)/out.config` file.
+- Main `Kconfig` file exposing configuration of modules.
+
+#### Output:
+- `$ (GENERATED_DIR)/autoconf.h` configuration header file.
+- Optional:
+    - `$ (GENERATED_DIR)/deps/*/*.h` header files that allow incremental builds
+
+
+### Summary of files
+These files are defined in `kconfig.mk`.
+
+| File           | Description |
+| ---------------| ----------- |
+| `Kconfig`      | Defines configuration options of modules. |
+| `Kconfig.dep`  | Holds a list of the modules that are being compiled. |
+| `app.config`   | Holds default application configuration values. |
+| `user.config`  | Holds configuration values applied by the user. |
+| `out.config`   | Configuration file containing all the symbols defined in `autoconf.h`. |
+| `out.config.d` | Dependency file of `out.config` containing the list of Kconfig files used to generate it. |
+| `autoconf.h`   | Header file containing the macros that applied the selected configuration. |
+
+## Kconfig symbols in Makefiles
+As '.config' files have Makefile syntax they can be included when building,
+which allows to access the applied configuration from the build system.
+
+During migration this is also useful, as it gives the ability to check if a
+parameter is being configured via Kconfig or a default value via `CFLAGS` could
+be injected. For example:
+
+```Makefile
+ifndef CONFIG_USB_VID
+  CFLAGS += -DCONFIG_USB_VID=0x1209
+endif
+```
+
+Symbols will have the same name as the configuration macros (thus will always
+have the `CONFIG_` prefix). As the configuration file is loaded in
+`Makefile.include` care should be taken when performing checks in the
+application's Makefile. The symbols will not be defined until after including
+`Makefile.include`.
+
+---
+# Transition phase
+
+## Making configuration via Kconfig optional
+During transition to the usage of Kconfig as the main configuration tool for
+RIOT, the default behavior will be the traditional one: expose configuration
+options in header files and use CFLAGS as inputs. To allow optional
+configuration via Kconfig, a convention will be used when writing Kconfig files.
+
+Modules should be contained in their own `menu` entries, this way the user
+can choose to enable the configuration via Kconfig for an specific module.
+These entries should define a dependency on the module they configure (see
+[Appendix A](#appendix-a-check-if-a-module-or-package-is-used) to see how to check if a module is being
+used).
+
+The module configuration must be enabled either via make dependency modelling.
```

```suggestion
The module configuration must be enabled via make dependency modelling.
```

I guess? The sentence would be missing the "or".